src/tests/unit/tests/injected/visualization/drawer.test.ts

```diff
@@ -877,35 +877,24 @@ describe('Drawer', () => {
 
         setupWindow();
         setupAddEventListerCalled(registerHandlerFunc);
-        testSubject.drawLayout();
 
-        // invoke scroll listener
-        const timeOutId = 10;
+        // draw
+        testSubject.drawLayout();
 
-        windowUtilsMock.setup(x => x.clearTimeout(It.isAny())).verifiable(Times.never());
-        windowUtilsMock
-            .setup(x => x.setTimeout(itIsFunction, HighlightBoxDrawer.recalculationTimeout))
-            .returns(() => timeOutId)
-            .verifiable();
+        const drawMock = Mock.ofInstance(() => {});
+        drawMock.setup(draw => draw()).verifiable(Times.exactly(2));
+        (testSubject as any).draw = drawMock.object;
 
+        // call once, run all timers, call again; should yield two draw calls.
```

Comment:
It would be nice for there to also be a test case that without `runAllTimers` in between, the throttling behavior causes `draw` to only execute once.

Reply:
Good call! I actually did have a test for that ... my concern was that it might potentially introduce flakiness to the tests as I was seeing inconsistent results. However, that was before I was using the mock timers and I think with them, it should be consistently failing.